### PR TITLE
update py.test to 2.9.2

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 3e8c615848242d5da64a3db5301e825c1915bdff:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by psav** [comment]
This one passed, but artifactor produced no output, it is a bit fickle on when things are setup and I wonder if the hookwrapper removal has had an effect. Is this a required change to go to newer pytest?

**Comment by RonnyPfannschmidt** [comment]
hookwrapper and historic hooks are incompatible
historic hooks redo all invocation on registration of new plugins,
a wrapper is semantically incompatible with such an invocation at a later point in time

**Comment by psav** [comment]
So then, I need to figure out on this branch what is going on with artifactor before it can be merged

File: fixtures/artifactor_plugin.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def pytest_addoption(parser):
                      help="A run id to assist in logging")
 
 
-@pytest.mark.hookwrapper
+@pytest.mark.tryfirst
 def pytest_configure(config):
     global proc
     if not SLAVEID and not proc and isinstance(art_client, ArtifactorClient):
```
Comment (psav) [inline fixtures/artifactor_plugin.py]: @RonnyPfannschmidt So I figured it out, there is a ```yield``` that needs to be removed below to make this work.

However then it crashes a little further on......I get this with pytest 2.9 also.

If I try to ```from fixtures.pytest_store import store``` I get an error like the following

Traceback (most recent call last):
  File "/home/cfme_tests/utils/artifactor_start.py", line 3, in <module>
    from artifactor import Artifactor, initialize
  File "/home/cfme_tests/artifactor/__init__.py", line 134, in <module>
    from utils.net import random_port
  File "/home/cfme_tests/utils/net.py", line 5, in <module>
    from fixtures.pytest_store import store
  File "/home/cfme_tests/fixtures/pytest_store.py", line 25, in <module>
    from _pytest.terminal import TerminalReporter
  File "/home/default/lib/python2.7/site-packages/_pytest/terminal.py", line 5, in <module>
    from _pytest.main import EXIT_OK, EXIT_TESTSFAILED, EXIT_INTERRUPTED, \
  File "/home/default/lib/python2.7/site-packages/_pytest/main.py", line 5, in <module>
    import pytest, _pytest
  File "/home/default/lib/python2.7/site-packages/pytest.py", line 27, in <module>
    _preloadplugins() # to populate pytest.* namespace so help(pytest) works
  File "/home/default/lib/python2.7/site-packages/_pytest/config.py", line 75, in _preloadplugins
    _preinit.append(get_config())
  File "/home/default/lib/python2.7/site-packages/_pytest/config.py", line 84, in get_config
    pluginmanager.import_plugin(spec)
  File "/home/default/lib/python2.7/site-packages/_pytest/config.py", line 384, in import_plugin
    __import__(importspec)
  File "/home/default/lib/python2.7/site-packages/_pytest/python.py", line 400, in <module>
    class PyCollector(PyobjMixin, pytest.Collector):
AttributeError: 'module' object has no attribute 'Collector'

**Comment by dajoRH** [comment]
Lint report for commit 67574bb45e4fe5558f27c07dc625ae5b61e4211b:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6da2a8c4464ffe5dbdffe3675eceaf4858c089ed:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6ded4603a0ad53af8d7ffab44bdf082180a12076:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d80f8f3d077853a0b35b5b68925ab9f58b478880:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by psav** [comment]
This seems to work fine for parallelized and singular runs - @amavinag we need to get a parallelized job on jenkins so that we can test this. It's a relatively big change, but a necessary one.

**Comment by dajoRH** [comment]
Lint report for commit 2d519888b18e8375387b2b7f40dc997d9b4f7ac1:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8feb727a94e4667ef1840a5c913c0f2d74b6328b:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d39b43cefcd3116c07a43c8829a6019d3862d71c:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b8de2016fed1e17a775a328f590960e8c338bcc7:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 60f163a859e4b946cfa0b5a966fb8dcf5a7a17d0:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 86e21611c708ca8723e5fa220851da82e2a840c2:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5049bfcbcf7396d1cab1fa32177859a55e3f1b60:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 240c0bf04b4e2a00cd829c721aba38fcdeacf056:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bdff299fea32679924f26adcdaa8f603d7b5f13a:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by jkrocil** [comment]
All the changes look reasonable to me :+1:

**Comment by dajoRH** [comment]
Lint report for commit 408066399fa2fe2bf782eb3d0023a2290fb398a0:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**